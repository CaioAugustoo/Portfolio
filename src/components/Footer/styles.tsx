import styled from "styled-components";

export const Footer = styled.footer`
  height: auto;
  width: 100%;
  margin-top: 120px;
`;

export const Rights = styled.p`
  font-size: 1.5rem;
  font-weight: 500;
  opacity: 0.7;
  text-align: center;
  margin: 0px 0 50px 0;
`;
