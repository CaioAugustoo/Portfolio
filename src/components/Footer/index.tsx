import React from "react";
import { Container } from "../../styles/globals";
import * as S from "./styles";

const Footer = () => {
  const currentDate = new Date();
  
  return (
    <S.Footer>
      <Container>
        <S.Rights>
          Â© ${currentDate.getFullYear()} - Caio Augusto <br />
          Alguns direitos reservados.
        </S.Rights>
      </Container>
    </S.Footer>
  );
};

export default Footer;
