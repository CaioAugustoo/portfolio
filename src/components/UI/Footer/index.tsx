import { LinksEnum } from "config/links";
import { Link } from "react-scroll";
import * as S from "./styles";

const Footer = () => {
  const currentDate = new Date().getFullYear();

  return (
    <S.Wrapper>
      <S.Links>
        <S.LinksWrapper>
          <li>
            <Link
              to="about"
              spy={true}
              smooth="easeInOutQuart"
              duration={1000}
              activeClass="active"
            >
              About
            </Link>
          </li>
          <li>
            <Link
              to="portfolio"
              spy={true}
              smooth="easeInOutQuart"
              duration={1000}
              activeClass="active"
            >
              Portfolio
            </Link>
          </li>
          <li>
            <a href={LinksEnum.Blog} target="_blank" rel="noopener">
              Blog
            </a>
          </li>
        </S.LinksWrapper>
      </S.Links>
      <S.Copyright>Â© {currentDate} - Caio Augusto</S.Copyright>
    </S.Wrapper>
  );
};

export default Footer;
