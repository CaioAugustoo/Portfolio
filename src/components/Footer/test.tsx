import { render, screen } from "@testing-library/react";

import Footer from "./index";

describe("<Footer />", () => {
  it("should render the texts", () => {
    render(<Footer />);

    expect(screen.getByText(/Â© 2020 - Caio Augusto/i)).toBeInTheDocument();
    expect(
      screen.getByText(/Alguns direitos reservados./i)
    ).toBeInTheDocument();
  });
});
