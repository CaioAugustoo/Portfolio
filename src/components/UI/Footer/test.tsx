import { screen } from "@testing-library/react";
import { renderWithTheme } from "utils/tests/helpers";

import Footer from ".";

describe("<Footer />", () => {
  it("should render correctly", () => {
    const { container } = renderWithTheme(<Footer />);

    expect(screen.getByText(/Â© 2021 - Caio Augusto/i)).toBeInTheDocument();
    expect(screen.getByText(/About/i)).toBeInTheDocument();
    expect(screen.getByText(/Portfolio/i)).toBeInTheDocument();
    expect(screen.getByRole("link", { name: /Blog/i })).toHaveAttribute(
      "href",
      "https://caio-blog.vercel.app/"
    );

    expect(container.firstChild).toMatchSnapshot();
  });
});
