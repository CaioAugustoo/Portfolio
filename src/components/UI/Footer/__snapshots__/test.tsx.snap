// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> should render correctly 1`] = `
.c0 {
  padding: 30px 0px;
  text-align: center;
  background-color: #0b0e18;
  margin: 15rem auto 0 auto;
}

.c1 a {
  font-size: 1.5rem;
  cursor: pointer;
  color: #ffffff;
}

.c1 a:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c2 li {
  margin: 10px;
}

.c3 {
  font-size: 1.4rem;
}

<footer
  class="c0"
>
  <nav
    class="c1"
  >
    <ul
      class="c2"
    >
      <li>
        <a>
          About
        </a>
      </li>
      <li>
        <a>
          Portfolio
        </a>
      </li>
      <li>
        <a
          href="https://caio-blog.vercel.app/"
          rel="noopener"
          target="_blank"
        >
          Blog
        </a>
      </li>
    </ul>
  </nav>
  <p
    class="c3"
  >
    Â© 
    2021
     - Caio Augusto
  </p>
</footer>
`;
